### PR TITLE
started on tests

### PR DESCRIPTION
**Pull request by chriscoey**
lots of tests to add
can also add integer conic tests etc with Pajarito later

will fail tests for now

### CODE REVIEW

**Comment by chriscoey** [comment]
@mlubin should we export everything? (then we don't have to prefix all the functions in test)

**Comment by mlubin** [comment]
@jrevels, any guidance with the exporting question? My understanding is that it's preferable not to export anything at all, but it would be nice for the tests to get everything in scope.

File: test/contconic.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        vc = addconstraint!(m, v, NonNegative(3))
+        c = addconstraint!(m, [3, 2], -[1 1 1; 0 1 1], Zero(2))
+        @test getattribute(m, ConstraintCount()) == 4
+
+        setattribute!(m, Sense(), MinSense)
+        setobjective!(m, 1, 0, v, [-3 -2, -4])
+    end
+
+    @testset "Solve model, check results" begin
+        optimize!(m)
```
Comment (mlubin) [inline test/contconic.jl]: I don't think ``m`` will be in scope here.

File: test/contlinear.jl [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        # min -x
+        # st   x + y <= 1   (x + y - 1 ∈ NonPositive)
+        #       x, y >= 0   (x, y ∈ NonNegative)
+
+        m = Model(solver)
+
+        v = addvariables!(m, 2)
+        @test getattribute(m, VariableCount()) == 2
+
+        @test cansetattribute(m, VariableLowerBound())
```
Comment (mlubin) [inline test/contlinear.jl]: We don't do ``VariableLowerBound`` anymore. We should add a variablewise constraint with the GreaterThan set.

File: test/runtests.jl [hunk 3]
```diff
@@ -1,5 +1,17 @@
-using MathOptInterface
-using Base.Test
+using MathOptInterface, Base.Test
+# TODO using solvers?
 
-# write your own tests here
-@test 1 == 1
+include("contlinear.jl")
+@testset "Continuous linear problems"
+    contlineartest(GLPKSolverLP())
```
Comment (mlubin) [inline test/runtests.jl]: No reason to list solvers here until they implement MOI

File: test/contconic.jl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        @test cangetattribute(m, TerminationStatus())
+        @test getattribute(m, TerminationStatus()) == Success
+
+        @test cangetattribute(m, PrimalStatus())
+        @test getattribute(m, PrimalStatus()) == FeasiblePoint
+        @test cangetattribute(m, DualStatus())
+        @test getattribute(m, DualStatus()) == FeasiblePoint
+
+        @test cangetattribute(m, ObjectiveValue())
+        @test isapprox(getattribute(m, ObjectiveValue()), -11, atol=eps)
```
Comment (joaquimg) [inline test/contconic.jl]: What if we use `@test getattribute(m, ObjectiveValue()) ≈ -11 atol=ε`?

Comment (chriscoey) [inline test/contconic.jl]: yup that seems nicer

Comment (mlubin) [inline test/contconic.jl]: We were using ``isapprox`` previously only for compatibility with 0.5.

**Comment by chriscoey** [comment]
OK i fixed those comments except for prefixing the functions, as the issue of whether to export is unresolved

**Comment by mlubin** [comment]
Let's not export things for now. Just put ``const MOI = MathOptInterface`` at the top of the tests and use the ``MOI.`` prefix.

**Comment by mlubin** [comment]
Also, tests need to be passing on travis before we can merge, otherwise the docs won't update.

**Comment by joaquimg** [comment]
Should we have a minimal TODO list to track what needs to accomplished before starting to work on solver APIs?

**Comment by jrevels** [comment]
> Just put const MOI = MathOptInterface at the top of the tests and use the MOI. prefix.

This kind of thing is generally my preference. Alternatively, you can just put the list of bindings you were going to export as the list of things you're `using` in the test namespace. Definitely feels like a bad idea to export everything.

**Comment by chriscoey** [comment]
Thanks Jarrett!

**Comment by chriscoey** [comment]
not updated for new functions. @mlubin perhaps you could update with how you intend the function syntax to be used then merge

File: test/contlinear.jl [hunk 5]
```diff
@@ -7,22 +7,22 @@ function contlineartest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64)
         # st   x + y <= 1   (x + y - 1 ∈ NonPositive)
         #       x, y >= 0   (x, y ∈ MOI.NonNegative)
```
Comment (joaquimg) [inline test/contlinear.jl]: Which one are we using in comments? I vote for not using `MOI.` in comments

Comment (mlubin) [inline test/contlinear.jl]: Yeah, we don't need MOI there

**Comment by chriscoey** [comment]
My bad, that was find and replace

**Pull request merged**